src/components/TodoApp.tsx

```diff
@@ -0,0 +1,110 @@
+import { useState } from 'react';
+import { useLocalStorage } from '../hooks/useLocalStorage';
+import { Todo } from '../types/Todo';
+import { TodoList } from './TodoList';
+import { TodosFilter } from './TodosFilter';
+
+export const TodoApp: React.FC = () => {
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+
+  const allCompletedTodos = todos.every(
+    (todo) => todo.completed,
+  );
+
+  const tickAllTodos = () => {
+    const checkAndTick = todos.map(todo => {
+      if (allCompletedTodos) {
+        return {
+          ...todo,
+          completed: false,
+        };
+      }
+
+      return {
+        ...todo,
+        completed: true,
+      };
```

Comment:
You have the boolean variable `allCompletedTodos`. So you can shorten this function
```suggestion
        return {
          ...todo,
          completed: !allCompletedTodos,
        };
```